python/tank/bootstrap/cached_configuration.py

```diff
@@ -71,6 +72,9 @@ def __init__(
 
         self._config_writer = ConfigurationWriter(self._path, self._sg_connection)
 
+        self.config_backup_folder_path = None
```

Comment:
Instance variables on a class are used to define the state of the object, e.g. what descriptor or project_id am I associated with for example. `self.config_backup_folder_path` does not represent such a state but you are merely using it to pass a value from one method to another. We should avoid this kind of pattern, better to pass it as method arguments if possible. Also, any private data which is not supposed to be used by any external classes should be prefixed with underscore, e.g. `self._config_backup_folder_path`. I suggest, when in doubt, look at all the other variables in the class (they all use underscores) and see if you can see a pattern that you can follow.